plugins/hdfs_assetstore/server/assetstore.py

```diff
@@ -129,17 +128,30 @@ def downloadFile(self, file, offset=0, headers=True):
             path = self._absPath(file)
 
         def stream():
-            skipped = 0
+            position = 0
             fileStream = self.client.cat([path]).next()
+            shouldBreak = False
             for chunk in fileStream:
-                if skipped < offset:
-                    if skipped + len(chunk) <= offset:
-                        skipped += len(chunk)
+                chunkLen = len(chunk)
+
+                if position < offset:
+                    if position + chunkLen <= offset:
+                        position += chunkLen
                     else:
-                        yield chunk[offset - skipped:]
-                        skipped = offset
+                        if position + chunkLen > endByte:
+                            chunkLen = endByte - position
+                            shouldBreak = True
+                        yield chunk[offset - position:chunkLen]
+                        position = offset + chunkLen - 1
```

Comment:
This chunk can be truncated at either the beginning or the end, so we'd have to take the beginning truncation into account in this block.